src/core/api/user.go

```diff
@@ -339,6 +341,57 @@ func (ua *UserAPI) ToggleUserAdminRole() {
 	}
 }
 
+// ListUserPermissions handles GET to /api/users/{}/permissions
+func (ua *UserAPI) ListUserPermissions() {
+	if ua.userID != ua.currentUserID {
+		log.Errorf("Current user, id: %d can not view other user's permissions", ua.currentUserID)
+		ua.RenderError(http.StatusForbidden, "User does not have permission")
+		return
+	}
+
+	relative := ua.Ctx.Input.Query("relative") == "true"
+
+	scope := rbac.Resource(ua.Ctx.Input.Query("scope"))
+	policies := []*rbac.Policy{}
+
+	namespace, err := scope.GetNamespace()
+	if err == nil {
```

Comment:
What about the err != nil ?

Reply:
`err == nil` is ok here, otherwise the code will be like this

```go
	namespace, err := scope.GetNamespace()
	if err != nil {
		ua.Data["json"] = []
		ua.ServeJSON()
		return
	}

	switch namespace.Kind() {
	case "project":
		for _, policy := range project.GetAllPolicies(namespace) {
			if ua.SecurityCtx.Can(policy.Action, policy.Resource) {
				policies = append(policies, policy)
			}
		}
	}

	results := []map[string]string{}
	for _, policy := range policies {
		var resource rbac.Resource

		// for resource `/project/1/repository` if `relative` is `true` then the resource in response will be `repository`
		if relative {
			relativeResource, err := policy.Resource.RelativeTo(scope)
			if err != nil {
				continue
			}
			resource = relativeResource
		} else {
			resource = policy.Resource
		}

		results = append(results, map[string]string{
			"resource": resource.String(),
			"action":   policy.Action.String(),
		})
	}

	ua.Data["json"] = results
	ua.ServeJSON()
	return
```

the following block is repeated
```go
	ua.Data["json"] = results
	ua.ServeJSON()
	return
```